mountpoint-s3-client/src/mock_client.rs

```diff
@@ -233,8 +233,9 @@ impl ObjectClient for MockClient {
         bucket: &str,
         key: &str,
         range: Option<Range<u64>>,
+        if_match: Option<String>,
     ) -> ObjectClientResult<Self::GetObjectResult, GetObjectError, Self::ClientError> {
-        trace!(bucket, key, ?range, "GetObject");
+        trace!(bucket, key, ?range, ?if_match, "GetObject");
```

Comment:
For later: we'll want the mock client to enforce this If-Match behavior too, so we can test it. The `add_object` method probably needs an `ETag` argument.